classes/gui/ScatterView.sc

```diff
@@ -1,46 +1,86 @@
-// 2005 by till bovermann
-//   bielefeld university
+/*********************************************************************
+(C) 2005 Till Bovermann / Bielefeld University
 
+and later...
+
+(C) 2018 Jonathan Reus / IEM Graz
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+    For more details see: https://www.gnu.org/licenses/
+**********************************************************************/
+
+
+/*
+@class ScatterView
+
+A simple 2-dimensional scatterplot with coordinate axes.
+
+@usage
+
+n = 100;
+x = Array.series(n, 0.0, 100 / n) + Array.fill(n, {rrand(-10.0, 10.0)}).round(0.01);
+y = Array.series(n, 1.0, 100 / n) + Array.fill(n, {rrand(-10.0, 10.0)}).round(0.01);
+p = x.collect {|item,i| [item, y[i]] }; // as points
+
+b = Rect(0,0, 800, 600);
+w = Window.new(bounds: b);
+j = ScatterView.new(w, b, p, [0,100].asSpec, [0,100].asSpec);
+j.drawAxis_(true).symbolColor_(Color.blue).drawMethod_(\fillOval).symbolSize_(5@5);
+w.front;
+
+*/
 ScatterView {
-	var <plot, <background;
+	var <plot, <>background;
 	var <>highlightColor, <highlightItem, <>highlightSize, <>isHighlight, <>drawAxis, <>drawValues;
 	var <>xAxisName, <>yAxisName;
 	var <symbolSize, <>symbolColor, <>drawMethod = \lineTo;
-	var adjustedData, specX, specY;
+	var <adjustedData, <>specX, <>specY;
```

Comment:
not so sure about adding getter/setter here... it at least needs a rerender, if you adjust the specs, no?

Reply:
Yup, I just realized that myself! I'm implementing a new method called setAxes to cover this functionality and will remove the setters... maybe it's easier to wait for me to do a new pull request once I've implemented that?